devito/mpi/routines.py

```diff
@@ -1108,11 +1118,11 @@ def _arg_values(self, args, **kwargs):
             for a in self.arguments:
                 if a.is_Dimension:
                     a_m, a_M = mapper[a]
-                    setattr(entry, a.min_name, mapper[a][0].subs(args))
-                    setattr(entry, a.max_name, mapper[a][1].subs(args))
+                    setattr(entry, a.min_name, _simple_subs(a_m, args))
+                    setattr(entry, a.max_name, _simple_subs(a_M, args))
```

Comment:
sorry, what's the difference with `a_m.subs(args)`, `a_M.subs(args)` here? Not sure I get it. I guess the `k in exprfree_symbols` prefilters ?

Reply:
Exactly.  Otherwise we spend a lot of time attempting to substitute in values that actually don't appear in the LHS - but sympy takes forever to figure that out.

Reply:
Can it be applied other places or is it the only critical one?

Reply:
I think we should apply it to profiling.py when we currently do `subs(args)`. I'll take a look later. We should also probably move this utility somewhere else, perhaps under symbolics